frappe/public/js/frappe/ui/notifications/notifications.js

```diff
@@ -325,6 +325,9 @@ class NotificationsView extends BaseNotificationsView {
 	}
 
 	get_item_link(notification_doc) {
+		if (notification_doc.custom_uri) {
+			return `/app/${notification_doc.custom_uri}`;
+		}
```

Comment:
```suggestion
		if (notification_doc.custom_uri) {
			return notification_doc.custom_uri;
		}
```


To keep this flexible in future, let's not put `/app` ourselves and expect custom code to add it.